crates/cairo-lang-protostar/src/lib.rs

```diff
@@ -36,3 +60,51 @@ pub fn build_protostar_casm_from_sierra(
     }
     Ok(Some(casm_contents))
 }
+
+pub fn setup_project_protostar(
+    db: &mut dyn SemanticGroup,
+    path: &Path,
+    crate_name: &str,
+) -> Result<Vec<CrateId>, ProjectError> {
+    if path.is_dir() {
+        match from_source_root_and_crate_name(path, crate_name) {
+            Ok(config) => {
+                let main_crate_ids = get_main_crate_ids_from_project(db, &config);
+                update_crate_roots_from_project_config(db, config);
+                Ok(main_crate_ids)
+            }
+            _ => Err(ProjectError::LoadProjectError),
+        }
+    } else {
+        Ok(vec![setup_single_file_project(db, path)?])
+    }
+}
+
+pub fn compile_path_protostar(
+    input_path: &str,
+    contract_path: Option<&str>,
+    compiler_config: CompilerConfig<'_>,
+    maybe_cairo_paths: Option<Vec<(&str, &str)>>,
```

Comment:
```suggestion
pub fn compile_from_resolved_dependencies(
    input_path: &str,
    contract_path: Option<&str>,
    compiler_config: CompilerConfig<'_>,
    linked_libraries: Vec<(&str, &str)>,
```
I would map option to empty vec in the bindings layer